File: ts/packages/knowPro/src/dateTimeSchema.ts
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

export type DateVal = {
    day: number;
    month: number;
    year: number;
};

export type TimeVal = {
    // In 24 hour form
    hour: number;
    minute: number;
    seconds: number;
};

export type DateTime = {
    date: DateVal;
    time?: TimeVal | undefined;
};

export type DateTimeRange = {
    startDate: DateTime;
    stopDate?: DateTime | undefined;
};

File: python/ta/typeagent/demo/date_time_schema.py
# Copyright (c) Microsoft Corporation.
# Licensed under the MIT License.

from pydantic.dataclasses import dataclass
from typing import Annotated
from typing_extensions import Doc


@dataclass
class DateVal:
    day: int
    month: int
    year: int


@dataclass
class TimeVal:
    hour: Annotated[int, Doc("In 24 hour form")]
    minute: int
    seconds: int


@dataclass
class DateTime:
    date: DateVal
    time: TimeVal | None = None


@dataclass
class DateTimeRange:
    start_date: DateTime
    stop_date: DateTime | None = None
